test/test_loops.py

```diff
@@ -84,6 +105,24 @@ def test_for_range_one_arg_loop(self):
         py_code = 'for i in range (5):\n\tprint(i)'
         self.assertEqual(js_code, main.main(py_code, 'py', 'js'))
 
+    def test_forOf(self):
+        js_code = 'for (elem of [1, 2]) {\n\t5\n}'
+        py_code = 'for elem in [1, 2]:\n\t5'
+        self.assertEqual(py_code, main.main(js_code, "js", "py"))
+        self.assertEqual(js_code, main.main(py_code, "py", "js"))
+
+    def test_forOf_with_java(self):
+        js_code = 'for (elem of arr) {\n\t5\n}'
+        py_code = 'for elem in arr:\n\t5'
+        java_code = 'for (int elem : arr) {\n\t5;\n}'
+
+        self.assertEqual(js_code, main.main(py_code, "py", "js"))
+        self.assertEqual(py_code, main.main(js_code, "js", "py"))
+        # Without context can't determine what type the elements in the array are,
```

Comment:
Maybe we used just a GenericType i.e. a fake class to indicate that we do not know the type?